rnd/autogpt_server/autogpt_server/server/ws_api.py

```diff
@@ -1,33 +1,89 @@
-from fastapi import WebSocket, WebSocketDisconnect
+import asyncio
+import logging
 
+from autogpt_libs.auth import auth_middleware, parse_jwt_token
+from fastapi import Depends, FastAPI, HTTPException, WebSocket, WebSocketDisconnect
+from fastapi.middleware.cors import CORSMiddleware
+
+from autogpt_server.data.queue import AsyncRedisEventQueue
+from autogpt_server.data.user import DEFAULT_USER_ID
 from autogpt_server.server.conn_manager import ConnectionManager
 from autogpt_server.server.model import ExecutionSubscription, Methods, WsMessage
+from autogpt_server.util.settings import Settings
 
+settings = Settings()
 
-async def websocket_router(websocket: WebSocket, manager: ConnectionManager):
-    await manager.connect(websocket)
-    try:
-        while True:
-            data = await websocket.receive_text()
-            message = WsMessage.model_validate_json(data)
-            if message.method == Methods.SUBSCRIBE:
-                await handle_subscribe(websocket, manager, message)
+app = FastAPI()
+event_queue = AsyncRedisEventQueue()
+_connection_manager = None
 
-            elif message.method == Methods.UNSUBSCRIBE:
-                await handle_unsubscribe(websocket, manager, message)
-            else:
-                print("Message type is not processed by the server")
-                await websocket.send_text(
-                    WsMessage(
-                        method=Methods.ERROR,
-                        success=False,
-                        error="Message type is not processed by the server",
-                    ).model_dump_json()
-                )
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=[
+        "*",
+    ],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
```

Comment:
IIRC allowing CORS from any origin is a potential security risk (cc @ntindle)

Reply:
Yes, we will get roasted again for it I bet 😞

Reply:
I just copied over what was in the app. 
re security risk, it's already blocked on the ingress level we only allow specific domains so its not _actually_ a security risk
but we can also change it here too

Reply:
We keep getting “roasted” by security researchers that don’t understand you can do it at different levels

Reply:
yep fixing it in this PR. my message was to also give context/knowledge share on this one